src/pages/[redirect]/index.tsx

```diff
@@ -0,0 +1,96 @@
+import fetchOriginalUrl from '@/utils/fetchOriginalUrl';
+import RedirectIcon from '../../../public/assets/icons/redirect';
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import Head from 'next/head';
+import { TINY_SITE } from '@/constants/url';
+
+const Redirect = () => {
+    const router = useRouter();
+    const { redirect: shortUrlCode } = router.query as { redirect: string };
+    const [originalUrl, setOriginalUrl] = useState('');
+    const [timer, setTimer] = useState(5);
+    const [isPremiumUser, setIsPremiumUser] = useState(false);
+    const [showTooltip, setShowTooltip] = useState(false);
+
+    useEffect(() => {
+        setIsPremiumUser(false);
+
+        if (shortUrlCode) {
+            fetchOriginalUrl(shortUrlCode)
+                .then((url) => {
+                    if (url) {
+                        setOriginalUrl(url);
+                    } else {
+                        console.log('Short URL code not found');
+                        router.push('/');
+                    }
+                })
+                .catch((error) => {
+                    console.error('Error fetching original URL:', error);
+                });
```

Comment:
can we use modern JavaScript feature async/await in place of promises. as its easier to read and maintain

Reply:
Done